obspy/clients/syngine/client.py

```diff
@@ -0,0 +1,602 @@
+            archive of SAC files, either ``miniseed`` or ``saczip``.
+        :type format: str
+        :param filename: Will download directly to the given file. If given,
+            this method will return nothing.
+        :type filename: str or file-like object
+        :param data: If given this will be sent directly sent to the Syngine
+            service as a POST payload. All other parameters except the
+            ``filename`` parameter will be silently ignored. Likely not that
+            useful for most people.
+        :type data: dictionary, bytes, or file-like object
+        """
+        # Send data straight via POST if given.
+        if data:
+            r = self._download(url=self._get_url("query"),
+                               data=data, filename=filename)
+            if filename:
+                return
+            return self.__read_to_stream(r=r)
+
+        if not bulk:
+            raise ValueError("Some bulk download information must be given.")
+
+        arguments = {
+            "eventid": eventid,
+            "sourcelatitude": sourcelatitude,
+            "sourcelongitude": sourcelongitude,
+            "sourcedepthinmeters": sourcedepthinmeters,
+            "sourcemomenttensor": sourcemomenttensor,
+            "sourcedoublecouple": sourcedoublecouple,
+            "sourceforce": sourceforce,
+            "origintime": origintime,
+            "starttime": starttime,
+            "endtime": endtime,
+            "label": label,
+            "components": components,
+            "units": units,
+            "scale": scale,
+            "dt": dt,
+            "kernelwidth": kernelwidth,
+            "format": format,
+            "filename": filename}
+
+        params = self._convert_parameters(model=model, **arguments)
+
+        # Assemble the bulk file.
+        with io.BytesIO() as buf:
+            # Write the header.
+            buf.write("model={model}\n".format(**params).encode())
+            del params["model"]
+            for key in sorted(params.keys()):
+                value = params[key]
+                buf.write("{key}={value}\n".format(key=key,
+                                                   value=value).encode())
+
+            # Write the bulk content.
+            for item in bulk:
+                # Dictionary like items.
+                if isinstance(item, collections.Mapping):
+                    if "latitude" in item or "longitude" in item:
+                        if not ("latitude" in item and "longitude" in item):
+                            raise ValueError(
+                                "Item '%s' in bulk must contain both "
+                                "latitude and longitude if either is given." %
+                                str(item))
+                        bulk_item = "{latitude} {longitude}".format(**item)
+                        for _i in ("netcode", "stacode", "loccode"):
```

Comment:
Yes it is a bit confusing. `networkcode` and `stationcode` can be used to associate network and station codes for events with custom receivers. `network` and `station` will actually be used to lookup the coordinates of real existing stations and use those for the coordinate lookup.

You are right for the second part. I'll change the bulk request to also take `networkcode`, `stationcode`, and `locationcode` parameters.